jazz_deployments/test/test.js

```diff
@@ -1,11 +1,1181 @@
+          responseObj.body = "{\"data\":{\"service\":\"mag!c\",\"domain\":\"k!ngd0m\",\"type\":\"api\"},\"input\":{\"service_id\":\"k!ngd0m_0f_mewni\"}}"
+          return obj.callback(null, responseObj, responseObj.body)
+        }
+      });
+      var reBuildDeployment = index.reBuildDeployment(event.body, config);
+      expect(reBuildDeployment.then((res) => {
+        reqStub.restore();
+        return res;
+      })).to.eventually.deep.equal(result);
+      sinon.assert.calledOnce(reqStub);
+    });
+  
+    it("should indicate error while re-building the deployment", () => {
+      var responseObj = {
+        statusCode: 400,
+        body: {
+          data: {},
+          message: err.message
+        }
+      },result = { 
+        result: 'success', 
+        message: 'deployment started.' 
+      };
+      reqStub = sinon.stub(request, "Request", (obj) => {
+        if(obj.method.toLowerCase() === 'post') {
+          return obj.callback(null, responseObj, responseObj.body)
+        } else if (obj.method === 'get') {
+          responseObj.body = "{\"data\":{\"service\":\"mag!c\",\"domain\":\"k!ngd0m\",\"type\":\"api\"},\"input\":{\"service_id\":\"k!ngd0m_0f_mewni\"}}"
+          return obj.callback(null, responseObj, responseObj.body)
+        }
+      });
+      var reBuildDeployment = index.reBuildDeployment(event.body, config);
+      expect(reBuildDeployment.then((res) => {      
+        return res;
+      })).to.be.rejectedWith(err.message);
+      sinon.assert.calledOnce(reqStub);
+      reqStub.restore();
+    });
+  });
 
-    // Add your test cases here.
-    assert(true);
-    done();
+  describe('handler', () => {
+    it("should indicate error during the generic validation",() => {
+      event.method = undefined;
+      message = '{"errorType":"BadRequest","message":"method cannot be empty"}';
+      index.handler(event, context,(err, res) => {
+          err.should.be.equal(message);
+          return err;
+      })
+    });
+  
+    it("should successfully create new deployment using POST method",() => {
+      event.method = "POST";
+      event.path =  {};
+      AWS.mock("DynamoDB.DocumentClient", "put", (params, cb) => {
+        return cb(null, dataObj);
+      });
+      index.handler(event, context,(err, res) => {
+          res.should.have.deep.property('data.deployment_id');
+          AWS.restore("DynamoDB.DocumentClient");
+          return res;
+      })
+    });
+  
+    it("should indicate error while creating new deployment using POST method",() => {
+      event.method = "POST";
+      event.path =  {};
+      message = '{"errorType":"InternalServerError","message":"unexpected error occurred"}';
+      AWS.mock("DynamoDB.DocumentClient", "put", (params, cb) => {
+        return cb(err, null);
+      });
+      index.handler(event, context,(err, res) => {
+        err.should.be.equal(message);
+        AWS.restore("DynamoDB.DocumentClient");
+        return res;
```

Comment:
why do you have to return here? remove return res from all the test where it is not used to assert later

Reply:
Removed